tests/playground.html

```diff
@@ -178,6 +179,65 @@
   }
 }
 
+function fakeDrag(id, dx, dy, opt_workspace) {
+  var ws = opt_workspace ? opt_workspace : Blockly.getMainWorkspace();
+  var blockToDrag = ws.getBlockById(id);
+
+  if (!blockToDrag) {
+    return;
+  }
+  var blockTop = blockToDrag.svgGroup_.getBoundingClientRect().top;
+  var blockLeft = blockToDrag.svgGroup_.getBoundingClientRect().left;
+
+  // Click somewhere on the block.
+  var mouseDownEvent = new MouseEvent('mousedown',
+      {clientX: blockLeft + 5, clientY: blockTop + 5});
+  blockToDrag.onMouseDown_(mouseDownEvent);
+
+  // Throw in a move for good measure.
+  window.setTimeout(
```

Comment:
Yes, to get a visible delay and to make sure it finishes rerendering before the next event, since the locations of connections depend on rendering.

If there's a more idiomatic way to do this I'm happy to change it--the nesting feels weird to me too.  I guess another option is to enqueue each event and then just dequeue with timeouts in between, instead of enqueueing objects saying which drags I want to do.